File: leetcode/medium/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum=Integer.MIN_VALUE;
        int n=nums.length;
        int sumn=0;
        for(int i=0;i<n;i++){
            sumn += nums[i];
            maxSum=Math.max(sumn,maxSum);
            if(sumn<0){
                sumn=0;
            }

        }
        return maxSum;
    }
}

File: leetcode/medium/0053-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int largestSum=INT_MIN;
        int sum=0;
        for(auto it : nums){
            sum+=it;
            largestSum=max(largestSum,sum);

            if(sum<0){
                sum=0;
            }
        }

        return largestSum;
    }
};